### PR TITLE
WID-78: PSE widget local launch

### PR DESCRIPTION
**Pull request by 915-Misan-Teodora**

### CODE REVIEW

**Review by popaula937 — COMMENTED** [review]
I would like to see more logs about the flow: when is a simulation starting, when has PSE finished execution, when is the H5 file written etc. These would give the user a bit more insight regarding the status of the PSE.

File: tvbwidgets/core/pse/parameters.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+                log.info(f"Reusing existing checkpoint dir {self.checkpoint_dir}")
+                # TODO consistency check
+            else:
+                os.mkdir(self.checkpoint_dir)
+                np.savetxt(os.path.join(self.checkpoint_dir, 'params.txt'), self.seq.params, fmt='%s')
+                np.save(os.path.join(self.checkpoint_dir, 'param_vals.npy'), self.seq.values)
+
+    def __call__(self, n_jobs=-1):
+        self._init_checkpoint()
+        from joblib import Parallel, delayed
```
Comment (popaula937) [inline tvbwidgets/core/pse/parameters.py:229]: Maybe move this import at the top of the file.

File: setup.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from pathlib import Path
 
 REQUIRED_PACKAGES = ["colorcet", "ebrains_drive", "ipympl>0.8.5", "ipywidgets==7.7.2", "mne>=1.0",
-                     "plotly==5.5.0", "pythreejs", "pyvista==0.37.0", "tvb-library>=2.5"]
+                     "plotly==5.5.0", "pythreejs", "pyvista==0.37.0", "tvb-library>=2.5", "tvb-framework"]
```
Comment (popaula937) [inline setup.py:14]: Add joblib as a dependency as well.

File: tvbwidgets/core/pse/parameters.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
+"""
```
Comment (liadomide) [inline tvbwidgets/core/pse/parameters.py:1]: Missing header. Pls copy from rest of modules.
Ideally also mention Jan Fousek as author, and the fact that this is a temporary copy from tvb-inversion package

File: tvbwidgets/ui/pse_launcher_widget.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        )
+        self.param_2 = widgets.Dropdown(
+            options=sorted(self.params_dict.keys()),
+            description=f"<b>PSE param2</b>",
+            value=list(sorted(self.params_dict.keys()))[2],
+            disabled=False
+        )
+
+        def param_changed(change):
+            if change['type'] != 'change' or change['name'] != 'value':
```
[the remark this answers is not in the document]
Comment (915-Misan-Teodora) [reply tvbwidgets/ui/pse_launcher_widget.py:148]: maybe a task for refactoring should be added. In more places the change of the trait is handled in the same way.

File: tvbwidgets/core/pse/parameters.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+            reduced = client.submit(reduction, metrics)
+            return reduced.result()
+        else:
+            return metrics
+
+
+def compute_metrics(sim, metrics):
+    computed_metrics = []
+    for metric in metrics:
+        if metric == "GlobalVariance":
```
[the remark this answers is not in the document]
Comment (915-Misan-Teodora) [reply tvbwidgets/core/pse/parameters.py:302]: The switch statement was added in python just in version 3.10 so for now the if case is good enough.

**Review by liadomide — APPROVED** [review]

**Pull request merged**